body.tex

```diff
@@ -0,0 +1,481 @@
+
+
+\subsection{Applicable Documents}
+
+\begin{tabbing}
+AUTH-NUM\= \kill
+\citeds{LDM-148} \>     DM Architecture\\
+\citeds{LDM-294} \>     DM Project Management Plan   \\
+\citeds{LDM-672} \>     LSST Software Release Management Policy \\
+% perhaps \citell{LL:AUTH-code}\>       Software Requirements Specification for \CU,\\
+\end{tabbing}
+
+
+\newpage
+\section{Definitions} \label{sec:definitions}
+
+This document uses certain terms which are often overloaded. In this section, we define these terms as intended in this document.
+
+
+\subsection{Environment} \label{sec:envdef}
+
+An environment is a set of libraries, executables, and configurations, that are predefined for a specific context or function.
+
+An environment is usually needed in the following contexts:
+
+\begin{itemize}
+\item software development: the environment includes all libraries and tools required to build and debug a software product
+\item software test and verification: the environment includes, on top of the build and debug tools, additional tools to permit/facilitate the testing and validation activities.
+\item operations: the environment shall include only tools required for the execution of an operational product. It shall not contain any build or debug tools. It shall be optimized for the operational activities. \footnote{In case a problem is found in operation the first thing to do is to replicate it in a different environment, where also debug and further analysis is possible.}
+\end{itemize}
+
+These are the most common uses of an environment. Additional scenarios can be identified.
+
+In some cases, the same environment is used in development, tests, and operations.
+
+Operational environment definitions, including location-specific configuration, shall never be included in the software products.
+Vanilla environment definition can be provided for unit tests purposes, as an example or template.
+
+
+\subsection{Software Product} \label{sec:swprod}
+
+A SW product is a \textit{defined} and \textit{controlled} software implementation with a specific scope, inside an overall system.
+
+\begin{itemize}
+\item \textit{defined}: it shall be clear which is the scope of the software product
+\item \textit{controlled}: the source code shall be maintained in a software repository, like for example GitHub.
+\end{itemize}
+
+%A software product is a component of the subsystem (DM) product tree.
+
+The following activities are done on a SW Product:
+\begin{itemize}
+\item development activities
+\item test activities, usually unit tests and system test
+\item release activities
+\item packaging activities
+\item operational activities, in standalone mode, in an operational pipeline or as part of a service. 
+Some SW Products are libraries, therefore are not operationally but used as dependencies by other software products.
+\end{itemize}
+
+A software product shall correspond to a single repository.
```

Comment:
Since this is a DMTN, I'd find the wording more clear if it were:

"With the exception of DM, a software product shall..." so that the DM readers know not to object when they read it.

Reply:
I would like to give here a general definition.
In fact, I have removed all references to DM implementation except this one.

I would prefer to keep it generic and move the following paragraph in a footnote.

We can also change the `shall` with a `should`, in order to make less strong this statement.

What do you think?